l10n_es_partner/models/res_partner.py

```diff
@@ -12,38 +12,59 @@ class ResPartner(models.Model):
     _inherit = "res.partner"
 
     comercial = fields.Char("Trade name", size=128, index=True)
-    display_name = fields.Char(compute="_compute_display_name")
 
+    # Extend original depends with comercial field
     @api.depends("comercial")
     def _compute_display_name(self):
         super()._compute_display_name()
 
+    @api.model
+    def _get_comercial_name_pattern(self):
+        return (
+            self.env["ir.config_parameter"]
+            .sudo()
+            .get_param("l10n_es_partner.name_pattern", default="")
+        )
+
     @api.model
     def search(self, args, offset=0, limit=None, order=None, count=False):
         """Include commercial name in direct name search."""
-        args = expression.normalize_domain(args)
-        for arg in args:
-            if isinstance(arg, (list, tuple)) and (
-                arg[0] == "name" or arg[0] == "display_name"
-            ):
-                index = args.index(arg)
-                args = (
-                    args[:index] + ["|", ("comercial", arg[1], arg[2])] + args[index:]
-                )
-                break
+        if "%(comercial_name)s" not in self._get_comercial_name_pattern():
+            args = expression.normalize_domain(args)
+            for arg in args:
+                if (
+                    isinstance(arg, (list, tuple))
+                    and (arg[0] == "name" or arg[0] == "display_name")
+                    and arg[2]
+                ):
+                    index = args.index(arg)
+                    args = (
+                        args[:index]
+                        + ["|", ("comercial", arg[1], arg[2])]
+                        + args[index:]
+                    )
+                    break
         return super().search(
             args, offset=offset, limit=limit, order=order, count=count
         )
 
     @api.model
-    def name_search(self, name, args=None, operator="ilike", limit=100):
+    def name_search(self, name="", args=None, operator="ilike", limit=100):
```

Comment:
This is not advised although Odoo does it, as the string is shared amongst different calls, and can cause side effects